[PB-4446]: Restructure common modules to top-level directories

terrerox opened the pull request:
## Description

  Reorganize shared code into a cleaner module structure by moving:
  - Shared components to src/components/
  - Core services to src/services/
  - Utilities to src/utils/
  - Hooks to src/hooks/
  
## Related Issues

<!-- Link any related GitHub issues "Fixes #<issue_number>" or "Relates to #<issue_number>". -->

## Related Pull Requests

<!-- List any related PRs in the format below:
- [Repository/Branch](link-to-PR)
-->

## Checklist

- [x] Changes have been tested locally.
- [x] Unit tests have been written or updated as necessary.
- [x] The code adheres to the repository's coding standards.
- [ ] Relevant documentation has been added or updated.
- [ ] No new warnings or errors have been introduced.
- [x] SonarCloud issues have been reviewed and addressed.
- [x] QA Passed

## Testing Process

<!-- Describe the testing process, including steps, configurations, and tools used to verify the changes. -->

## Additional Notes

<!-- Include any additional context, potential impacts, or implementation details that reviewers should be aware of. -->

sg-gs commented:
Unless @CandelR, this is too much @terrerox. Please follow the convention of less than 500 LOCs per PR. You can move it in part

terrerox committed:
refactor: restructure common modules to top-level directories

  Reorganize shared code into a cleaner module structure by moving:
  - Shared components to src/components/
  - Core services to src/services/
  - Utilities to src/utils/
  - Hooks to src/hooks/

terrerox committed:
refactor: update import paths for common module restructure

Update import paths in app/, analytics, auth, banners, core, crypto, database, drive, i18n, network, and payment modules to reflect the new top-level directory structure for common modules, services, and components.

terrerox committed:
refactor: align shared imports with new components/services aliases

Point repositories, share flows, Redux slices, workers, and Drive/Home/Backups/NewSettings views to the relocated modules under components/* and services/*, replacing the legacy app/shared and app/core/services paths. Ensure related Signup, Trash, Recents, PublicShared, and Settings tests/hooks compile against the consolidated structure introduced by the shared module reorganization.

terrerox committed:
refactor: consolidate shared components and service tests

Move BaseInput/Skeleton into components/, drop the unused captcha helper, update affected share/storage/signup/trash specs to the new services/* aliases, and add fresh desktop, operating-system, and screen service tests to match the module layout.

terrerox committed:
test: expand coverage for shared services and utils

Add new Vitest suites for drag-and-drop, validation, captcha token generation, queue utilities, and the network service helper to lock in behavior after the common-module restructuring.

terrerox committed:
test: add coverage for stream and zip services

Create new Vitest suites for stream.service and zip.service, ensuring download/queue handling and archive helpers are validated against the reorganized shared service layer.

terrerox committed:
test: add unit specs for error and socket services

Add Vitest suites for the shared error and socket services to validate error reporting, websocket lifecycle, and event handling expected after the common-module restructure.

terrerox committed:
test: add unit coverage for workspace service

Introduce a vitest suite for src/services/workspace.service.ts, mocking SDK responses and validating team/workspace data flows to safeguard behavior under the new shared services structure.

xabg2 commented on src/components/OnBoardingModal.tsx:
```diff
@@ -78,10 +78,11 @@ export const OnboardingModal: FC<OnBoardingModalProps> = ({ onCloseModalPressed
             data-test="download-desktop-modal-close-button"
           >
-            <img src={XSVG} />
+            <img src={XSVG} alt="" />
```
What about this `alt`?

xabg2 commented on src/utils/backupKeyUtils.ts:
```diff
@@ -99,7 +92,7 @@ export const detectBackupKeyFormat = (
     }
   } catch (err) {
-    // Not JSON, might be an old format (just plain mnemonic)
+    errorService.reportError(err);
```
Not sure if we want to report the error as it can be caused by an old format 🤔

terrerox committed:
test: improve drag-and-drop test descriptions for clarity

terrerox committed:
test: improve test descriptions for error and socket services

terrerox committed:
test: remove redundant singleton test from QueueUtilsService

terrerox committed:
test: remove redundant singleton test from workspace.service

terrerox committed:
refactor: simplify error handling and improve accessibility

terrerox committed:
fix: use crypto alias for browser compatibility

terrerox committed:
refactor: remove unused network service and fix captcha test

terrerox committed:
refactor: remove unused screen service

terrerox committed:
refactor: remove unused network service export from index

terrerox committed:
refactor: streamline service imports and improve test structure

terrerox committed:
Merge branch 'master' into refactor/common-module-structure

terrerox committed:
refactor: update import path for localStorageService and remove unused BaseInput component

terrerox committed:
refactor: update mocks to use vi.importActual for improved test accuracy

terrerox committed:
Merge branch 'master' into refactor/common-module-structure
